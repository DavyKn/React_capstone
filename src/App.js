import { Routes, Route } from 'react-router-dom';
import Navbar from './components/nav/NavBar';
/* eslint-disable */
import CoinDetails from './Pages/CoinDetails';
import Coins from './Pages/Coins';
/* eslint-enable */

function App() {
  return (
    <div className="App">
      <Navbar />

      <Routes>
        <Route path="/" element={<Coins />} />
        <Route path="/coindetail" element={<CoinDetails />} />
      </Routes>
    </div>
  );
}

export default App;
