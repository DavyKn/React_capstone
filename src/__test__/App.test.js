import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
/* eslint-disable */
import store from '../redux/store';
import App from '../App';
/* eslint-enable */

describe('Coins test', () => {
  test('snapshot for coins', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <App />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
