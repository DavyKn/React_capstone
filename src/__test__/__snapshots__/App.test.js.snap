// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coins test snapshot for coins 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <nav
          class="navigation-bar"
        >
          <div
            class="nav-logo"
          >
            <ul
              class="navigation-ul"
            >
              <li
                class="NavLink"
              >
                <a
                  aria-current="page"
                  class="logo active"
                  href="/"
                >
                  <img
                    alt="Home logo"
                    class="prism"
                    src="home.png"
                  />
                </a>
              </li>
            </ul>
          </div>
          <div
            class="logo-div"
          >
            <p
              class="logo-p"
            >
              Crypto Currencies
            </p>
          </div>
        </nav>
        <div>
          <div
            class="div-search"
          >
             
            <input
              class="search"
              placeholder="Search coin"
              type="search"
              value=""
            />
          </div>
          <ul
            class="coin-container"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <nav
        class="navigation-bar"
      >
        <div
          class="nav-logo"
        >
          <ul
            class="navigation-ul"
          >
            <li
              class="NavLink"
            >
              <a
                aria-current="page"
                class="logo active"
                href="/"
              >
                <img
                  alt="Home logo"
                  class="prism"
                  src="home.png"
                />
              </a>
            </li>
          </ul>
        </div>
        <div
          class="logo-div"
        >
          <p
            class="logo-p"
          >
            Crypto Currencies
          </p>
        </div>
      </nav>
      <div>
        <div
          class="div-search"
        >
           
          <input
            class="search"
            placeholder="Search coin"
            type="search"
            value=""
          />
        </div>
        <ul
          class="coin-container"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
